crates/kubelet/src/node.rs

```diff
@@ -389,3 +383,120 @@ fn lease_spec_definition(node_name: &str) -> serde_json::Value {
         }
     )
 }
+
+/// Defines the labels that will be applied to this node
+///
+/// Default values and passed node-labels arguments are injected by config.
+fn node_labels_definition(arch: &str, config: &Config) -> HashMap<String, String> {
+    // Add mandatory static labels
+    let mut labels = HashMap::new();
+    labels.insert("beta.kubernetes.io/os".to_owned(), "linux".to_owned());
+    labels.insert("kubernetes.io/os".to_owned(), "linux".to_owned());
+    labels.insert("kubernetes.io/role".to_owned(), "agent".to_owned());
+    labels.insert("type".to_owned(), "krustlet".to_owned());
+    // add the mandatory labels that are dependent on injected values
+    labels.insert("beta.kubernetes.io/arch".to_owned(), arch.to_owned());
+    labels.insert("kubernetes.io/arch".to_owned(), arch.to_owned());
+    labels.insert(
+        "kubernetes.io/hostname".to_owned(),
+        config.hostname.to_owned(),
+    );
+
+    let k8s_namespace = "kubernetes.io";
+    // namespaces managed by this method - do not allow user injection
+    let managed_namespace_labels = vec![
+        "beta.kubernetes.io/arch",
+        "beta.kubernetes.io/os",
+        "kubernetes.io/arch",
+        "kubernetes.io/hostname",
+        "kubernetes.io/os",
+        "kubernetes.io/role",
+        "type",
+    ];
+    let allowed_k8s_namespace_labels = vec![
```

Comment:
Since this list doesn't need to grow any bigger, you can just make it a fixed sized array by dropping the `vec!` part of the declaration.

Reply:
applied and pushed